packages/fluentui/react-northstar/src/components/Dialog/Dialog.tsx

```diff
@@ -229,8 +229,7 @@ class Dialog extends AutoControlledComponent<WithAsProp<DialogProps>, DialogStat
     // and ESC is pressed, the opened Dialog should get closed and the trigger should get focus
     const lastOverlayRef = getRefs().pop();
     const isLastOpenedDialog: boolean = lastOverlayRef && lastOverlayRef.current === this.overlayRef.current;
-
-    if (keyboardKey.getCode(e) === keyboardKey.Escape && isLastOpenedDialog) {
+    if (keyboardKey.getCode(e) === keyboardKey.Escape && isLastOpenedDialog && this.overlayRef.current === e.target) {
```

Comment:
As was mentioned in chat this will not work as `this.overlayRef.current === e.target` will be always `false`

Reply:
Solved